extensions/mssql/src/sparkFeature/dialog/sparkJobSubmission/sparkJobSubmissionDialog.ts

```diff
@@ -0,0 +1,162 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as sqlops from 'sqlops';
+import * as vscode from 'vscode';
+import * as nls from 'vscode-nls';
+import * as utils from '../../../utils';
+import * as LocalizedConstants from '../../../localizedConstants';
+
+import { AppContext } from '../../../appContext';
+import { ApiWrapper } from '../../../apiWrapper';
+import { SparkJobSubmissionModel } from './sparkJobSubmissionModel';
+import { SparkConfigurationTab } from './sparkConfigurationTab';
+import { SparkJobSubmissionInput } from './sparkJobSubmissionService';
+import { SparkAdvancedTab } from './sparkAdvancedTab';
+
+const localize = nls.loadMessageBundle();
+
+export class SparkJobSubmissionDialog {
+    private _dialog: sqlops.window.modelviewdialog.Dialog;
+    private _dataModel: SparkJobSubmissionModel;
+    private _sparkConfigTab: SparkConfigurationTab;
+    private _sparkAdvancedTab: SparkAdvancedTab;
+    private get apiWrapper(): ApiWrapper {
+        return this.appContext.apiWrapper;
+    }
+
+    constructor(
+        private sqlClusterConnection: sqlops.connection.Connection,
```

Comment:
Minor: couldn't we just be passing around the actual `SqlClusterConnection` class to these? Would simplify things a lot and reduce confusion since we shouldn't need to define a Connection/IConnectionProfile anywhere the Knox endpoint is used.